storage/postgres/tree_storage.go

```diff
@@ -393,3 +403,52 @@ func (t *treeTX) Close() error {
 	}
 	return nil
 }
+
+func (t *treeTX) GetMerkleNodes(ctx context.Context, treeRevision int64, nodeIDs []storage.NodeID) ([]storage.Node, error) {
+	return t.subtreeCache.GetNodes(nodeIDs, t.getSubtreesAtRev(ctx, treeRevision))
+}
+
+func (t *treeTX) SetMerkleNodes(ctx context.Context, nodes []storage.Node) error {
```

Comment:
Fine to include in a follow up (or just pop a `TODO` in somewhere), but we just added enforcement of query serialisation on a Tx  - should probably do the same in this impl at some point too (see #1567 )

Reply:
I will get to this one.

Reply:
:+1: